### ISSUE TITLE
Problem with back-pressure 'STOR' code 

### ISSUE THREAD

**Issue by oleksiyk** [sstur/nodeftpd#89]
Its hard to replicate but sometimes `storeStream` fails with error 'write after end'. That `.end()` is called here: https://github.com/sstur/nodeftpd/blob/master/lib/FtpConnection.js#L1214 

I guess it happens when the peer has finished writing all the data but the dataSocket was paused. The 'drain' event of the storeStream in this case has two handlers: https://github.com/sstur/nodeftpd/blob/master/lib/FtpConnection.js#L1201
and
https://github.com/sstur/nodeftpd/blob/master/lib/FtpConnection.js#L1201
The first handler calls `resume()` on a dataSocket but the second handler closes the storeStream with `end()`.

Maybe someone has some quick fix idea?

**Comment by oleksiyk** [comment]
I wonder is there a reason not to use `pipe`:

```javascript
function handleUpload(dataSocket) {
    dataSocket.pipe(storeStream);
    dataSocket.on('error', function(err) {
      notErr = false;
      self._logIf(LOG.ERROR, 'Data connection error: ' + util.inspect(err));
    });
  }
```

`pipe` internally handles all the overwhelming.

**Comment by sstur** [comment]
Thanks for tracking down this "write after end" error! I've seen it before, but wasn't too sure what's causing it. You seem to have diagnosed it pretty well.

As for the fix, I'm not sure why we were manually piping, I assume there's some custom logic in there so let me take a closer look before merging your diff. Thanks!

**Comment by oleksiyk** [comment]
Closed with #90
